multiwrapper/multiwrapper.go

```diff
@@ -0,0 +1,193 @@
+package multiwrapper
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	sync "sync"
+
+	wrapping "github.com/hashicorp/go-kms-wrapping/v2"
+)
+
+const baseEncryptor = "__base__"
+
+var _ wrapping.Wrapper = (*MultiWrapper)(nil)
+
+var ErrKeyNotFound = errors.New("given key ID not found")
+
+// MultiWrapper allows multiple wrappers to be used for decryption based on key
+// ID. This allows for rotation of data by allowing data to be decrypted across
+// multiple (possibly derived) wrappers and encrypted with the default.
+// Functions on this type will likely panic if the wrapper is not created via
+// NewMultiWrapper.
+type MultiWrapper struct {
```

Comment:
Very pedantic, but i was a bit confused by this name. I expected this to encrypt a plaintext N times with N different wrappers. Perhaps `MultiUnwrapper`?

Reply:
I'll rename it.

Reply:
I think maybe what I'll do is change the module to `multi` and change this to `PooledWrapper`. I think your idea of encrypting multiple times with multiple wrappers is interesting and from that perspective may want to reserve MultiWrapper/MultiUnwrapper names for that purpose. Sound good?

Reply:
That name works for me! Thanks